### PR TITLE
Code clean up and add support for passing the request through the whole token generation lifecycle.

### PR DESCRIPTION
**Pull request by jeffreydwalter**
@pkieltyka sorry for the post merge PR, but I realized I needed to add support for passing the request object through all of the validation and claims/props funcs to allow someone to add something to the request context in the validation step which could then be added as a claim or property if needed. I needed this functionality so I could add some claims dynamically based on the user/secret used when requesting an oauth token.

I also went through the code again and cleaned it up more...

### CODE REVIEW

File: example/authserver/main.go [hunk 1]
```diff
@@ -77,47 +75,50 @@ type TestUserVerifier struct {
...
+	return "", nil
 }
 
 // AddClaims provides additional claims to the token
-func (*TestUserVerifier) AddClaims(ctx context.Context, tokenType oauth.TokenType, credential, tokenID, scope string) (map[string]string, error) {
+func (*TestUserVerifier) AddClaims(tokenType oauth.TokenType, credential, tokenID, scope string, r *http.Request) (map[string]string, error) {
 	claims := make(map[string]string)
-	claims["customerId"] = "1001"
-	claims["customerData"] = `{"OrderDate":"2016-12-14","OrderId":"9999"}`
+	claims["customer_id"] = "1001"
```
Comment (pkieltyka) [inline example/authserver/main.go:103]: In JS world / json, the convention is to use camelCase as it was before, but both are fine..

Comment (jeffreydwalter) [reply example/authserver/main.go:103]: Per the standard, JSON doesn't impose any convention on keys. I think the conventions are more determined by authors of the API.
Facebook and Twitter use snake case for their JSON in their APIs:
- https://developers.facebook.com/docs/graph-api/field-expansion/
- https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/overview
Google uses camel case:
- https://google.github.io/styleguide/jsoncstyleguide.xml

 I prefer snake case, but am happy to do whatever for these examples.

Comment (jeffreydwalter) [reply example/authserver/main.go:103]: Also, all of the oAuth structs for this package (and the gin version) use snake case. 🤷🏻

Comment (jeffreydwalter) [reply example/authserver/main.go:103]: @pkieltyka can we merge this or are you not going to until I update those JSON keys?

**Pull request merged**